assembly/pallets/balances/balances.ts

```diff
@@ -43,30 +51,96 @@ export class Balances {
      * @param value value of the transfer
      */
     static transfer(source: AccountIdType, dest: AccountIdType, value: Balance): u8[] {
-        Log.info("trying to transfer: " + value.toString());
-        const senderAccData = BalancesStorageEntries.Account().get(source);
-        const receiverAccData = BalancesStorageEntries.Account().get(dest);
-
-        const info = SystemStorageEntries.Account().get(source);
-        const nonce = info.nonce;
+        const senderBalance = BalancesStorageEntries.Account().get(source);
+        const receiverBalance = BalancesStorageEntries.Account().get(dest);
 
-        if(senderAccData.getFree().unwrap() < value.unwrap()) {
+        if(senderBalance.unwrap() < value.unwrap()) {
             return ResponseCodes.INSUFFICIENT_BALANCE;
         }
 
-        const senderNewBalance = senderAccData.getFree().unwrap() - value.unwrap();
-        const receiverNewBalance = receiverAccData.getFree().unwrap() + value.unwrap();
+        const senderNewBalance = senderBalance.unwrap() - value.unwrap();
+        const receiverNewBalance = receiverBalance.unwrap() + value.unwrap();
 
+        // Make sure new balances is higher than ExistentialDeposit
         if(senderNewBalance < BalancesConfig.existentialDeposit().unwrap()) {
             return ResponseCodes.VALIDITY_ERROR;
         }
         
-        this.setBalance(source, instantiate<Balance>(senderNewBalance), senderAccData.getReserved());
-        this.setBalance(dest, instantiate<Balance>(receiverNewBalance), receiverAccData.getReserved());
+        this.setBalance(source, instantiate<Balance>(senderNewBalance), instantiate<Balance>(0));
+        this.setBalance(dest, instantiate<Balance>(receiverNewBalance), instantiate<Balance>(0));
 
-        info.setNonce(instantiate<NonceType>(nonce.unwrap() + 1));
-        SystemStorageEntries.Account().set(info);
-        Log.info("Done transfering: " + value.toString());
+        // Increment nonce
+        const info = SystemStorageEntries.Account().get(source);
+        info.setNonce(instantiate<NonceType>(info.nonce.unwrap() + 1));
+        SystemStorageEntries.Account().set(info, source);
+        Balances._depositEvent(EventTypes.Transfer, source.toU8a().concat(dest.toU8a()).concat(value.toU8a()));
+        
         return ResponseCodes.SUCCESS;
     }
+
+    static _depositEvent(name: i32, args: u8[]): void {

```

Comment:
I think that what we could do is have `System_depositEvent(key, type)` where `key` is the `Storage.get(Utils.getHashedKey("Balances", "Events", null));` and type is `BALANCESET` or `TRANSFER`